examples/source_examples/awardco/connector.py

```diff
@@ -0,0 +1,103 @@
+"""This connector fetches user data from the AwardCo API and upserts it into the destination using the Fivetran Connector SDK.
+See the Technical Reference documentation (https://fivetran.com/docs/connectors/connector-sdk/technical-reference#update)
+and the Best Practices documentation (https://fivetran.com/docs/connectors/connector-sdk/best-practices) for details
+"""
+
+import json
+from pathlib import Path
+from fivetran_connector_sdk import Connector
+from fivetran_connector_sdk import Logging as log
+from fivetran_connector_sdk import Operations as op
+import requests
+
+
+def validate_configuration(configuration: dict):
+    """
+    Validate the configuration dictionary to ensure it contains all required parameters.
+    Args:
+        configuration: a dictionary that holds the configuration settings for the connector.
+    Raises:
+        ValueError: if any required configuration parameter is missing.
+    """
+    required_configs = ["api_key", "base_url"]
+    for key in required_configs:
+        if key not in configuration:
+            raise ValueError(f"Missing required configuration value: {key}")
+
+
+def schema(configuration: dict):
+    """
+    Define the schema function which lets you configure the schema your connector delivers.
+    Args:
+        configuration: a dictionary that holds the configuration settings for the connector.
+    """
+    return [
+        {
+            "table": "user",
+            "primary_key": ["employeeId"],
+            "columns": {
+                "employeeId": "STRING",
+                "firstName": "STRING",
+                "lastName": "STRING",
+                "email": "STRING",
+                "balance": "FLOAT",
+                "currencyCode": "STRING",
+            },
+        },
+    ]
+
+
+def update(configuration: dict, state: dict):
+    """
+    Define the update function, which is called by Fivetran during each sync.
+    Args:
+        configuration: A dictionary containing connection details
+        state: A dictionary containing state information from previous runs
+    """
+    log.warning("Starting data sync from AwardCo API")
+
+    validate_configuration(configuration=configuration)
+
+    api_key = configuration.get("api_key")
+    base_url = configuration.get("base_url")
+    use_mock = str(configuration.get("use_mock", "false")).strip().lower() in {
+        "1",
+        "true",
+        "yes",
+        "y",
+    }
+    last_sync_time = state.get("last_sync_time")
```

Comment:
We can have a default time if there is no last_sync_time in state in the required format
```suggestion
    last_sync_time = state.get("last_sync_time","1990-01-01T00:00:00")
```
Check the format of the datetime string and change it as required